SSA/Core/EffectKind.lean

```diff
@@ -15,8 +17,8 @@ def toMonad (e : EffectKind) (m : Type → Type) : Type → Type :=
 
 section Lemmas
 
-@[simp] theorem toMonad_pure    : pure.toMonad m = Id := rfl
-@[simp] theorem toMonad_impure  : impure.toMonad m = m := rfl
+@[simp, simp_denote] theorem toMonad_pure   : pure.toMonad m = Id := rfl
+@[simp, simp_denote] theorem toMonad_impure : impure.toMonad m = m := rfl
```

Comment:
Can we think whether we need both `simp` as well as `simp_denote`?

Reply:
So this definitely has to be `simp_denote`, for the automation to work.

Besides that, I feel it makes sense as a global `simp` lemma. When writing a proof by hand, I'd expect `pure.toMonad m` to simplify, so, to me, it'd be surprising if `simp` (without extra prompting) didn't do so. Especially since not all `simp_denote` lemmas necessarily make sense in a manual proof context, so I wouldn't want to have to rely on `simp [simp_denote]`.